n_t_with_s/Snakefile

```diff
@@ -0,0 +1,386 @@
+
+try:
+    mask_file = config["mask_file"]
+except KeyError:
+    mask_file = None
+
+# Run full pipeline!
+rule all:
+   input: output_dir + f"/Results/all_estimated_Ne_{selection}.pdf"
+
+## Run only the simulations!
+#rule all:
+#    input: expand(output_dir + "/Intermediate/{seeds}/{chrms}.trees", 
+#                    seeds=seed_array, chrms=chrm_list) 
+
+## Run only stairwayplot and plot with coalescent rate!
+#rule all:
+#   input: expand(output_dir + '/Results/stairwayplot_{selection}.pdf', selection=selection)
+
+## Run only smc+++ 
+#rule all:
+#    input: expand(output_dir + "/Intermediate/{seeds}/{chrms}_{selection}.trees.smc.gz", seeds=seed_array, chrms=chrm_list, selection=selection) 
+
+## Run only mcmc
+#rule all:
+#    input: expand(output_dir+"/Results/msmc_estimated_Ne_{selection}.png", selection=selection)
+
+rule download_genetic_map:
+    output: genetic_map_downloaded_flag
+    message: "Downloading default genetic map"
+    run:
+        # We need to have this here to avoid several threads trying to download the 
+        # the genetic map into the cache at the same time.
+        if genetic_map_id is not None:
+            genetic_map = species.get_genetic_map(genetic_map_id)
+            if not genetic_map.is_cached():
+                genetic_map.download()
+            with open(output[0], "w") as f:
+                print("File to indicate genetic map has been downloaded", file=f)
+
+
+rule simulation:
+    input: 
+        genetic_map_downloaded_flag,
+        # rules.top_directory.output,
+    output:
+        output_dir + "/Intermediate/{seeds}/{chrms}_{selection}.trees"
+    # TODO get simulation parameters from stdpopsim, so we can share them easily 
+    # with anlysis code below?
+    threads: 1 
+    run:
+        contig = species.get_contig(wildcards.chrms, genetic_map=genetic_map_id)
+        samples = model.get_samples(*num_samples_per_population)
+        engine = stdpopsim.get_engine("slim") ## here we use the slim engine
+        ####### Here we are adding a given DFE
+        ####### Adding DFE/selection to the simulation  
+        dfe = species.get_dfe(dfe_id)
+        selection = wildcards.selection
+
+        if selection == "exons":
+            ## Adding annotation only seletion on exon region
+            exons = species.get_annotations(annotation_id)
+            exon_intervals = exons.get_chromosome_annotations(wildcards.chrms)
+            contig.add_dfe(intervals=exon_intervals, DFE=dfe)
+
+        if selection == "all":
+            # Adding selection to the whole contig
+            contig.add_dfe(intervals=np.array([[0, int(contig.length)]]), DFE=dfe)
+
+        if selection == "none":
+            # if selection is not specified then a neutral model is expected
+            contig = species.get_contig(wildcards.chrms, genetic_map=genetic_map_id)
+
+        ts = engine.simulate(model, contig, samples, seed=wildcards.seeds, slim_scaling_factor=1,
+            slim_burn_in=10)
```

Comment:
please add `slim_scaling_factor` to the `config.yaml` so this could be
```suggestion
        ts = engine.simulate(
            model, 
            contig, 
            samples, 
            seed=wildcards.seeds, 
            slim_scaling_factor=config["slim_scaling_factor),
            slim_burn_in=10
            )
```

Reply:
added.